op-dispute-mon/mon/claims.go

```diff
@@ -84,62 +82,56 @@ func (c *ClaimMonitor) checkUpdateHonestActorStats(proxy common.Address, claim *
 
 func (c *ClaimMonitor) checkGameClaims(
 	game *types.EnrichedGameData,
-	claimStatus map[metrics.ClaimStatus]int,
+	claimStatuses *metrics.ClaimStatuses,
 	honest map[common.Address]*metrics.HonestActorData,
 ) {
 	// Check if the game is in the first half
-	duration := uint64(c.clock.Now().Unix()) - game.Timestamp
+	now := c.clock.Now()
+	duration := uint64(now.Unix()) - game.Timestamp
 	firstHalf := duration <= game.MaxClockDuration
 
+	minDescendantAccumulatedTimeByIndex := make(map[int]time.Duration)
+
 	// Iterate over the game's claims
-	for _, claim := range game.Claims {
+	// Reverse order so we can track whether the claim has unresolvable children
+	for i := len(game.Claims) - 1; i >= 0; i-- {
+		claim := game.Claims[i]
 		c.checkUpdateHonestActorStats(game.Proxy, &claim, honest)
 
 		// Check if the clock has expired
 		if firstHalf && claim.Resolved {
-			c.logger.Error("Claim resolved in the first half of the game duration", "game", game.Proxy, "claimContractIndex", claim.ContractIndex, "id", claim.ID(), "clock", duration)
+			c.logger.Error("Claim resolved in the first half of the game duration", "game", game.Proxy, "claimContractIndex", claim.ContractIndex, "clock", duration)
 		}
 
 		maxChessTime := time.Duration(game.MaxClockDuration) * time.Second
 		var parent faultTypes.Claim
 		if !claim.IsRoot() {
 			parent = game.Claims[claim.ParentContractIndex].Claim
 		}
-		accumulatedTime := faultTypes.ChessClock(c.clock.Now(), claim.Claim, parent)
+		accumulatedTime := faultTypes.ChessClock(now, claim.Claim, parent)
+
+		// Calculate the minimum accumulated time of this claim or any of its descendants
+		minAccumulatedTime, ok := minDescendantAccumulatedTimeByIndex[claim.ContractIndex]
+		if !ok || accumulatedTime < minAccumulatedTime {
+			minAccumulatedTime = accumulatedTime
+		}
+		// Update the minimum accumulated time for the parent claim to include this claim's time.
+		curr, ok := minDescendantAccumulatedTimeByIndex[claim.ParentContractIndex]
```

Comment:
Shouldn't this be looking at descendants that are on the same depth modulo 2?
A parent claim's can be resolvable but not its direct child if its clock is greater than the child and  there aren't any other descendants.

Reply:
The claim can't be resolved until all its descendants are resolved. We're not trying to track the time left on the clock for this particular "team" in this case, but just identify how long since the claim first became resolvable. Hence we don't do the modulo 2 thing and are looking at direct children (which since we're iterating backwards then rolls up to provide the minimum accumulated time for any descendant).

Reply:
I see. Thanks for explaining.